modules/prediction/evaluator/vehicle/lane_scanning_evaluator.cc

```diff
@@ -101,6 +105,20 @@ void LaneScanningEvaluator::Evaluate(Obstacle* obstacle_ptr,
                  latest_feature_ptr);
 }
 
+bool LaneScanningEvaluator::ExtractStringFeatures(
+    const LaneGraph& lane_graph,
+    std::vector<std::string>* const string_feature_values) {
+  for (int i = 0; i < lane_graph.lane_sequence_size(); ++i) {
+    const LaneSequence& lane_sequence = lane_graph.lane_sequence(i);
+    string_feature_values->push_back("|");
+    for (int j = 0; j < lane_sequence.lane_segment_size(); ++j) {
+      const LaneSegment& lane_segment = lane_sequence.lane_segment(j);
```

Comment:
Learn the `for-each` style:
`for (const LaneSegment& lane_segment : lane_sequence.lane_segment())`

Use it whenever possible. It is simpler and could avoid off-one bug